ksql-engine/src/main/java/io/confluent/ksql/services/KafkaTopicClientImpl.java

```diff
@@ -279,11 +292,36 @@ private static boolean isTopicDeleteEnabled(final AdminClient adminClient) {
           .orElse(true);
 
     } catch (final Exception e) {
-      log.error("Failed to initialize TopicClient: {}", e.getMessage());
+      LOG.error("Failed to initialize TopicClient: {}", e.getMessage());
       throw new KsqlException("Could not fetch broker information. KSQL cannot initialize", e);
     }
   }
 
+  private static Config getConfig(final AdminClient adminClient) {
+    try {
+      final Collection<Node> brokers = adminClient.describeCluster().nodes().get();
+      final Node broker = Iterables.getFirst(brokers, null);
+      if (broker == null) {
+        LOG.warn("No available broker found to fetch config info.");
+        throw new KsqlServerException(
+            "AdminClient discovered an empty Kafka Cluster. "
+                + "Check that Kafka is deployed and KSQL is properly configured.");
+      }
+
+      final ConfigResource configResource = new ConfigResource(Type.BROKER, broker.idString());
+
+      final Map<ConfigResource, Config> brokerConfig = ExecutorUtil.executeWithRetries(
```

Comment:
Arguably outside the scope of this PR, but wouldn't it be nice if this looped around the available brokers when retrying...

Reply:
🤔 that would be quite interesting... I'll group that with the PR described above.